packages/introspect/src/generate.ts

```diff
@@ -180,7 +188,11 @@ function createTableJsonSchema(
         // In the future we should add support for composite types.
         const type = isComposite
             ? {}
-            : { $ref: `#/$defs/${createTypeDefId(column.type)}` };
+            : isExpandedTypeAnArray(column.expandedType)
```

Comment:
Let's move this logic to a function with a regular if/else. 3 cases is already enough.